File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum=0;
        int total=0;
        for(int num:nums)
            total+=num;
        for(int i=0;i<nums.length;sum+=nums[i++])
        {
            if(sum*2==total-nums[i])
                return i;
        }
        return -1;
    }
}

File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int left_sum[nums.size()];
        int right_sum[nums.size()];
        int sum=0;
        left_sum[0]=0;
        for(int i=1;i<nums.size();i++)
        {
            left_sum[i]=nums[i-1]+left_sum[i-1];
        }
        right_sum[nums.size()-1]=0;
        for(int i=nums.size()-2;i>=0;i--)
        {
            right_sum[i]=nums[i+1]+right_sum[i+1];
        }
        for(int i=0;i<nums.size();i++)
        {
            if(left_sum[i]==right_sum[i])
            {
                return i;
            }
        }
        return -1;
    }
};